File: VB/CustomizeExportAction.Web/WebApplication.vb
Imports Microsoft.VisualBasic
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Xpo
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports DevExpress.ExpressApp.Web

Namespace CustomizeExportAction.Web
   Partial Public Class CustomizeExportActionAspNetApplication
	   Inherits WebApplication
		Protected Overrides Sub CreateDefaultObjectSpaceProvider(ByVal args As CreateCustomObjectSpaceProviderEventArgs)
			args.ObjectSpaceProvider = New XPObjectSpaceProvider(args.ConnectionString, args.Connection)
		End Sub
	  Private module1 As DevExpress.ExpressApp.SystemModule.SystemModule
	  Private module2 As DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule
	  Private module3 As CustomizeExportAction.Module.CustomizeExportActionModule
	  Private module4 As CustomizeExportAction.Module.Web.CustomizeExportActionAspNetModule
	  Private module6 As DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule
	  Private module5 As DevExpress.ExpressApp.Validation.ValidationModule

	  Public Sub New()
		 InitializeComponent()
	  End Sub

	  Private Sub CustomizeExportActionAspNetApplication_DatabaseVersionMismatch(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs) Handles MyBase.DatabaseVersionMismatch
			e.Updater.Update()
			e.Handled = True
	  End Sub

	  Private Sub InitializeComponent()
		 Me.module1 = New DevExpress.ExpressApp.SystemModule.SystemModule()
		 Me.module2 = New DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule()
		 Me.module3 = New CustomizeExportAction.Module.CustomizeExportActionModule()
		 Me.module4 = New CustomizeExportAction.Module.Web.CustomizeExportActionAspNetModule()
		 Me.module5 = New DevExpress.ExpressApp.Validation.ValidationModule()
		 Me.module6 = New DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule()
		 CType(Me, System.ComponentModel.ISupportInitialize).BeginInit()
		 ' 
		 ' module5
		 ' 
		 Me.module5.AllowValidationDetailsAccess = True
		 ' 
		 ' CustomizeExportActionAspNetApplication
		 ' 
		 Me.ApplicationName = "CustomizeExportAction"
		 Me.Modules.Add(Me.module1)
		 Me.Modules.Add(Me.module2)
		 Me.Modules.Add(Me.module6)
		 Me.Modules.Add(Me.module3)
		 Me.Modules.Add(Me.module4)
		 Me.Modules.Add(Me.module5)
'		 Me.DatabaseVersionMismatch += New System.EventHandler(Of DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs)(Me.CustomizeExportActionAspNetApplication_DatabaseVersionMismatch);
		 CType(Me, System.ComponentModel.ISupportInitialize).EndInit()

	  End Sub
   End Class
End Namespace

File: CS/CustomizeExportAction.Web/WebApplication.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.ExpressApp.Web;

namespace CustomizeExportAction.Web {
   public partial class CustomizeExportActionAspNetApplication : WebApplication {
      private DevExpress.ExpressApp.SystemModule.SystemModule module1;
      private DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule module2;
      private CustomizeExportAction.Module.CustomizeExportActionModule module3;
      private CustomizeExportAction.Module.Web.CustomizeExportActionAspNetModule module4;
      private DevExpress.ExpressApp.Security.SecurityModule securityModule1;
      private DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule module6;
      private DevExpress.ExpressApp.Validation.ValidationModule module5;

      public CustomizeExportActionAspNetApplication() {
         InitializeComponent();
      }

      private void CustomizeExportActionAspNetApplication_DatabaseVersionMismatch(object sender, DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs e) {
            e.Updater.Update();
            e.Handled = true;
      }

      private void InitializeComponent() {
         this.module1 = new DevExpress.ExpressApp.SystemModule.SystemModule();
         this.module2 = new DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule();
         this.module3 = new CustomizeExportAction.Module.CustomizeExportActionModule();
         this.module4 = new CustomizeExportAction.Module.Web.CustomizeExportActionAspNetModule();
         this.module5 = new DevExpress.ExpressApp.Validation.ValidationModule();
         this.module6 = new DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule();
         this.securityModule1 = new DevExpress.ExpressApp.Security.SecurityModule();
         ((System.ComponentModel.ISupportInitialize)(this)).BeginInit();
         // 
         // module5
         // 
         this.module5.AllowValidationDetailsAccess = true;
         // 
         // CustomizeExportActionAspNetApplication
         // 
         this.ApplicationName = "CustomizeExportAction";
         this.Modules.Add(this.module1);
         this.Modules.Add(this.module2);
         this.Modules.Add(this.module6);
         this.Modules.Add(this.securityModule1);
         this.Modules.Add(this.module3);
         this.Modules.Add(this.module4);
         this.Modules.Add(this.module5);
         this.DatabaseVersionMismatch += new System.EventHandler<DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs>(this.CustomizeExportActionAspNetApplication_DatabaseVersionMismatch);
         ((System.ComponentModel.ISupportInitialize)(this)).EndInit();

      }
   }
}
